### PR TITLE
feat: add option to disable specific conversions

### LINKED ISSUE #212: Disable automatic conversions
Disable conversions implemented by Mapperly via the `MapperAttribute`.

Relates https://github.com/riok/mapperly/discussions/211

### PR DESCRIPTION
**Pull request by latonz**
Closes #212

### CODE REVIEW

File: src/Riok.Mapperly.Abstractions/MappingConversionType.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    /// <summary>
+    /// None.
+    /// </summary>
+    None = 0,
+
+    /// <summary>
+    /// Use a target type constructor
+    /// which has a single parameter of the type of the source.
+    /// </summary>
+    Ctor = 1 << 0,
```
Comment (CommonGuy) [inline src/Riok.Mapperly.Abstractions/MappingConversionType.cs:19]: Would write this out as `Constructor`.

Maybe also change the description to
> Use the constructor of the target type, which accepts the source type as a single parameter.

File: src/Riok.Mapperly.Abstractions/MappingConversionType.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    /// <summary>
+    /// If the source type is a <see cref="string"/>,
+    /// uses a a static visible method named `Parse` on the target type
+    /// with a return type equal to the target type and a string as single parameter.
+    /// </summary>
+    ParseMethod = 1 << 3,
+
+    /// <summary>
+    /// If the target type is a <see cref="string"/>,
+    /// uses the `ToString` on the source type.
```
Comment (CommonGuy) [inline src/Riok.Mapperly.Abstractions/MappingConversionType.cs:40]: missing word "method" after "ToString

File: src/Riok.Mapperly.Abstractions/MappingConversionType.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    /// If the source is an <see cref="Enum"/>
+    /// and the target is a <see cref="string"/>,
+    /// uses the name of the enum member to convert it to a string.
+    /// </summary>
+    EnumToString = 1 << 6,
+
+    /// <summary>
+    /// If the source is an <see cref="Enum"/>
+    /// and the target is another <see cref="Enum"/>,
+    /// depending on the <see cref="EnumMappingStrategy"/>
```
Comment (CommonGuy) [inline src/Riok.Mapperly.Abstractions/MappingConversionType.cs:61]: Would simplify this to
```
If the source is an <see cref="Enum"/> and the target is another <see cref="Enum"/>, map it according to the <see cref="EnumMappingStrategy"/>.
```

File: src/Riok.Mapperly.Abstractions/MappingConversionType.cs (hunk 1 above)
Comment (latonz) [reply src/Riok.Mapperly.Abstractions/MappingConversionType.cs]: Improved.

File: src/Riok.Mapperly.Abstractions/MappingConversionType.cs (hunk 2 above)
Comment (latonz) [reply src/Riok.Mapperly.Abstractions/MappingConversionType.cs]: Improved.

File: src/Riok.Mapperly.Abstractions/MappingConversionType.cs (hunk 3 above)
Comment (latonz) [reply src/Riok.Mapperly.Abstractions/MappingConversionType.cs]: Improved.

**Review by CommonGuy — APPROVED** [review]

**Pull request merged**